Source/JavaScriptCore/API/glib/JSCValue.cpp

```diff
@@ -2145,3 +2145,33 @@ char* jsc_value_to_json(JSCValue* value, unsigned indent)
 
     return json;
 }
+
+/**
+ * jsc_value_new_promise:
+ * @context: a #JSCContext
+ * @resolve: (out) (transfer full) (nullable): a callable #JSCValue.
+ * @reject: (out) (transfer full) (nullable): a callable #JSCValue.
+ *
+ * Create a new Promise returning handles to resolve and reject method
+ *
+ * Returns: (transfer full): a deferred promise object
+ *
+ * Since: 2.48
+ */
+JSCValue* jsc_value_new_promise(JSCContext* context, JSCValue **resolve, JSCValue **reject)
+{
+    g_return_val_if_fail(JSC_IS_CONTEXT(context), nullptr);
+    auto* jsContext = jscContextGetJSContext(context);
+    JSObjectRef resolveObj = nullptr;
+    JSObjectRef rejectObj = nullptr;
+    JSValueRef exception = nullptr;
+    JSCValue* ret = jscContextGetOrCreateValue(context, JSObjectMakeDeferredPromise(jsContext, resolve ? &resolveObj : nullptr, reject ? &rejectObj : nullptr, &exception)).leakRef();
+    if (jscContextHandleExceptionIfNeeded(context, exception))
+        return nullptr;
+
+    if (resolve)
+        *resolve = jscContextGetOrCreateValue(context, resolveObj).leakRef();
+    if (reject)
+        *reject = jscContextGetOrCreateValue(context, rejectObj).leakRef();
+    return ret;
```

Comment:
What about promise rejected by an exception? in case of exception the promise is rejected and the exception can be handled in catch(), and not exposed to the context. That's why I suggested to use callback that receives resolve/reject. That way we can push/pop a function call context to catch exceptions. See the [Objc API](https://developer.apple.com/documentation/javascriptcore/jsvalue/init(newpromisein:fromexecutor:)?changes=lat____4&language=objc)